packages/rich-text/src/plugins/Hyperlink/HyperlinkModal.tsx

```diff
@@ -234,16 +228,29 @@ interface HyperLinkDialogData {
 
 export async function addOrEditLink(
   editor: ReactEditor & HistoryEditor & SPEditor,
-  { linkText, linkTarget, linkType, sdk }: addOrEditLinkProps
+  sdk: FieldExtensionSDK
 ) {
   if (!editor.selection) return;
 
+  let linkType;
+  let linkText;
+  let linkTarget;
+  let linkEntity;
+
+  const [node, path] = getNodeEntryFromSelection(editor, LINK_TYPES);
+  if (node) {
+    linkType = node.type;
+    linkText = (node.children[0] as TextElement).text;
+    linkTarget = (node.data as { uri: string }).uri || '';
+    linkEntity = (node.data as { target: Link }).target;
+  }
+
   const selectionBeforeBlur = { ...editor.selection };
-  const currentLinkText = linkText ?? Editor.string(editor, editor.selection);
+  const currentLinkText = linkText || Editor.string(editor, editor.selection);
```

Comment:
This is confusing me with line 243. I think we might not need both, what I mean is, I think we can simply use `Editor.string(editor, editor.selection)` for both because `editor.selection` should return what's selected, regardless of which component is selected. In this case, we only need the text, regardless of the action (edit or insert).

Reply:
We're capturing a couple different cases here.

1) *_Create_ case:* the link doesn't exist yet. As a user I want to highlight one small selected part of a text node and convert that into a link. For this I need the `editor.selection` so the whole node isn't selected.

2) *_Update_ case:* the link exists. As a user, I want to click on it to prompt the edit modal. I haven't actually _selected_ anything, I just clicked on the link. So I need the `path` of the clicked node to get the link text, since the `editor.selection` buffer will possibly be empty or unrelated to the link at hand.

Reply:
Ah, that's true! You're right.